controllers/sr_controller/sr_controller.py

```diff
@@ -127,21 +142,17 @@ def main():
 
     print("Using {} for Zone {}".format(robot_file, robot_zone))
 
-    env = os.environ.copy()
-    # Ensure the python path is properly passed down so the `sr` module can be imported
-    env['PYTHONPATH'] = os.pathsep.join(sys.path + [str(ROOT / "modules")])
-    env['SR_ROBOT_ZONE'] = str(robot_zone)
-    env['SR_ROBOT_MODE'] = robot_mode
-    env['SR_ROBOT_FILE'] = str(robot_file)
-
-    completed_process = subprocess.run(
-        [sys.executable, "-u", str(robot_file)],
-        env=env,
-        cwd=str(robot_file.parent),
-    )
-
-    # Exit with the same return code so webots reports it as an error
-    sys.exit(completed_process.returncode)
+    # Pass through the various data our library needs
+    os.environ['SR_ROBOT_ZONE'] = str(robot_zone)
+    os.environ['SR_ROBOT_MODE'] = robot_mode
+    os.environ['SR_ROBOT_FILE'] = str(robot_file)
+
+    # Swith to running the competitor code
+    reconfigure_environment(robot_file)
+
+    robot_globals = {}  # type: Dict[str, object]
```

Comment:
Any reason to define this upfront rather than passing in 2 constructed dicts to the `exec` call below?

Reply:
I misread the behaviour of `exec` that we needed to pass the same instance to both `globals` and `locals` in order to get the behaviour we want. Having re-checked that we could pass this inline.

However the other reason I pulled this out is that we can (in theory) use this to override which builtins are available. Maybe not worth having it separate for now.

Reply:
da6a9ed